packages/auth/src/grant/routes.ts

```diff
@@ -233,7 +233,7 @@ async function getGrantDetails(
   }
   const { id: interactId, nonce } = ctx.params
   const grant = await grantService.getByInteractionSession(interactId, nonce)
-  if (!grant) {
+  if (!grant || grant.state === GrantState.Revoked) {
```

Comment:
Just a suggestion: if we want to keep this check in the service layer, we could pass an optional arg to `getByInteractionSession`, something like `includeRevoked` (false by default). We could also include `.whereNot('state', 'REVOKED')` in cases where it is false in the DB lookup.

Reply:
That might make more sense. In most cases `GrantState.Revoked` is treated the same as the grant not existing, but there are a few places it's explicitly handled. This would apply to `getByContinue` as well.

Reply:
i implemented this on `getByContinue` and `getByInteractionSession`